NPE when `jbang config list --global --verbose --show-origin`

**Describe the bug**
When listing configuration a NPE is raised if `--global` flag is used.

If `--global` is not used, the configuration is displayed without any error.

**To Reproduce**
Steps to reproduce the behavior:
1. With JBang freshly installed
2. jbang config --verbose set run.debug 7272
3. jbang config list --global --verbose --show-origin
4. See error

**Expected behavior**
Configuration is displayed without NPE.

**JBang version**
jbang version
0.90.2

**Additional context**
java_user@linux_box:~$ jbang version
0.90.2

java_user@linux_box:~$ jbang config --verbose set run.debug 7272
[jbang] jbang version 0.90.2
[jbang] Writing configuration to /home/java_user/.jbang/jbang.properties
[jbang] Option 'run.debug' set to '7272' in /home/java_user/.jbang/jbang.properties

java_user@linux_box:~$ jbang config list --global --verbose --show-origin
[jbang] jbang version 0.90.2
[jbang] [ERROR] Cannot invoke "dev.jbang.source.ResourceRef.getOriginalResource()" because the return value of "dev.jbang.Configuration.getStoreRef()" is null
java.lang.NullPointerException: Cannot invoke "dev.jbang.source.ResourceRef.getOriginalResource()" because the return value of "dev.jbang.Configuration.getStoreRef()" is null
	at dev.jbang.cli.ConfigList.printConfigWithOrigin(Config.java:191)
	at dev.jbang.cli.ConfigList.doCall(Config.java:158)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:93)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:15)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
	at dev.jbang.cli.JBang$3.handle(JBang.java:145)
	at dev.jbang.cli.JBang$3.handle(JBang.java:140)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at dev.jbang.Main.main(Main.java:14)
[jbang] If you believe this a bug in jbang, open an issue at https://github.com/jbangdev/jbang/issues

java_user@linux_box:~$ jbang config list --verbose --show-origin
[jbang] jbang version 0.90.2
/home/java_user/.jbang/jbang.properties
   run.debug = 7272
classpath:/jbang.properties
   init.template = hello
   run.debug = 4004
   run.jfr = filename={baseName}.jfr
   wrapper.install.dir = .

--- a/src/main/java/dev/jbang/Configuration.java
+++ b/src/main/java/dev/jbang/Configuration.java
@@ -188,7 +188,7 @@ public class Configuration {
 			if (cfgFileName != null) {
 				Path cfgFile = Util.getCwd().resolve(cfgFileName);
 				if (Files.isReadable(cfgFile)) {
-					global = read(cfgFile);
+					global = get(cfgFile);
 				} else {
 					global = defaults();
 				}
@@ -218,6 +218,17 @@ public class Configuration {
 			defaults = Configuration.getBuiltin();
 		}
 		return defaults;
+	}
+
+	public static Configuration get(Path catalogPath) {
+		return get(ResourceRef.forNamedFile(catalogPath.toString(), catalogPath.toFile()));
+	}
+
+	private static Configuration get(ResourceRef ref) {
+		Path configPath = ref.getFile().toPath();
+		Configuration cfg = read(configPath);
+		cfg.storeRef = ref;
+		return cfg;
 	}
 
 	/**

--- a/src/main/java/dev/jbang/cli/BaseCommand.java
+++ b/src/main/java/dev/jbang/cli/BaseCommand.java
@@ -36,7 +36,7 @@ public abstract class BaseCommand implements Callable<Integer> {
 	@CommandLine.Option(names = { "--config" }, description = "Path to config file to be used instead of the default")
 	void setConfig(Path config) {
 		if (Files.isReadable(config)) {
-			Configuration.instance(Configuration.read(config));
+			Configuration.instance(Configuration.get(config));
 		} else {
 			warn("Configuration file does not exist or could not be read: " + config);
 		}

--- a/src/main/java/dev/jbang/cli/Config.java
+++ b/src/main/java/dev/jbang/cli/Config.java
@@ -34,7 +34,7 @@ abstract class BaseConfigCommand extends BaseCommand {
 	protected Configuration getConfig(Path cwd, boolean strict) {
 		Path cfgFile = getConfigFile(strict);
 		if (cfgFile != null) {
-			return Configuration.read(cfgFile);
+			return Configuration.get(cfgFile);
 		} else {
 			return Configuration.getMerged();
 		}

--- a/src/test/java/dev/jbang/TestConfiguration.java
+++ b/src/test/java/dev/jbang/TestConfiguration.java
@@ -1,9 +1,7 @@
 package dev.jbang;
 
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.emptyOrNullString;
-import static org.hamcrest.Matchers.equalTo;
-import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.Matchers.*;
 
 import java.io.IOException;
 import java.nio.file.Files;
@@ -16,6 +14,7 @@ import dev.jbang.cli.Edit;
 import dev.jbang.cli.Init;
 import dev.jbang.cli.JBang;
 import dev.jbang.cli.Run;
+import dev.jbang.source.ResourceRef;
 import dev.jbang.util.Util;
 
 import picocli.CommandLine;
@@ -80,6 +79,16 @@ public class TestConfiguration extends BaseTest {
 		assertThat(cfg2.get("foo"), equalTo("abc"));
 		assertThat(cfg2.get("bar"), equalTo("ghi"));
 		assertThat(cfg2.get("baz"), equalTo("jkl"));
+	}
+
+	@Test
+	public void testReadAndGetConfig() throws IOException {
+		Path cfgFile = jbangTempDir.resolve(Configuration.JBANG_CONFIG_PROPS);
+		Configuration cfgRead = Configuration.read(cfgFile);
+		assertThat(cfgRead.getStoreRef(), nullValue());
+		Configuration cfgGet = Configuration.get(cfgFile);
+		assertThat(cfgGet.getStoreRef(), notNullValue());
+		assertThat(cfgGet.getStoreRef(), equalTo(ResourceRef.forResource(cfgFile.toString())));
 	}
 
 	@Test

